### PR TITLE
REVSDL-2379 Mobile command factory for VR Plugin

### PR DESCRIPTION
**Pull request by thinhts**
add mobile command factory for VR Plugin
fix load plugin error
@giangtqh 
@KKarlash 
@bonewell 
@DKlimenko 
@VladSemenyuk

### CODE REVIEW

**Comment by VladSemenyuk** [comment]
Reviewed

File: src/components/vr_cooperation/src/base_command_request.cc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "vr_cooperation/commands/base_command_request.h"
+
```
Comment (bonewell) [inline src/components/vr_cooperation/src/base_command_request.cc]: `
#include "utils/logger.h"
`

File: src/components/vr_cooperation/src/register_service_request.cc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "vr_cooperation/commands/register_service_request.h"
+
```
Comment (bonewell) [inline src/components/vr_cooperation/src/register_service_request.cc]: `
#include "utils/logger.h"
`

File: src/components/vr_cooperation/src/vr_module.cc [hunk 3]
```diff
@@ -37,6 +37,7 @@ namespace vr_cooperation {
 
 using functional_modules::ProcessResult;
 
+EXPORT_FUNCTION_IMPL(VRModule);
```
Comment (bonewell) [inline src/components/vr_cooperation/src/vr_module.cc]: Without Singleton you might to use another macro. See IVDCM plugin solution.

**Comment by KKarlash** [comment]
Please extend your factory for UnregisterService and ActivateService requests.

**Comment by DKlimenko** [comment]
Reviewed

File: src/components/vr_cooperation/src/vr_module.cc (hunk 3 above)
Comment (thinhts) [inline src/components/vr_cooperation/src/vr_module.cc]: we will fix it when we implement VRModule

**Comment by thinhts** [comment]
Please review
@giangtqh 
@KKarlash 
@bonewell 
@DKlimenko 
@VladSemenyuk

File: src/components/vr_cooperation/src/mobile_command_factory.cc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "vr_cooperation/mobile_command_factory.h"
+
+namespace vr_cooperation {
+
+commands::Command* MobileCommandFactory::CreateCommand(
+    const application_manager::MessagePtr& msg) {
+
```
Comment (bonewell) [inline src/components/vr_cooperation/src/mobile_command_factory.cc]: I'm afraid that cpplint will report warning about unnecessary empty line.

**Comment by bonewell** [comment]
Reviewed.

File: src/components/vr_cooperation/src/mobile_command_factory.cc (hunk 4 above)
Comment (thinhts) [inline src/components/vr_cooperation/src/mobile_command_factory.cc]: No, It isn't.

File: src/components/vr_cooperation/include/vr_cooperation/commands/active_service_request.h [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+#include "vr_cooperation/commands/base_command_request.h"
+
+namespace vr_cooperation {
+
+namespace commands {
+
+/**
+ * @brief ActiveServiceRequest command class
+ */
+class ActiveServiceRequest : public BaseCommandRequest {
```
Comment (KKarlash) [inline src/components/vr_cooperation/include/vr_cooperation/commands/active_service_request.h]: Activ**at**eServiceRequest

**Comment by KKarlash** [comment]
reviewed

**Pull request merged**